react/features/base/util/uri.js

```diff
@@ -96,6 +96,44 @@ function _fixURIStringScheme(uri: string) {
     return uri;
 }
 
+/**
+ * Converts a room name to a backend-safe format. Properly lowercased and url encoded.
+ *
+ * @param {string?} room - The room name to convert.
+ * @returns {string?}
+ */
+export function getBackendSafeRoomName(room: ?string): ?string {
+    if (!room) {
+        return room;
+    }
+
+    /* eslint-disable no-param-reassign */
+    try {
+        // We do not know if we get an already encoded string at this point
+        // as different platforms do it differently, but we need a decoded one
+        // for sure. However since decoding a non-encoded string is a noop, we're safe
+        // doing it here.
+        room = decodeURIComponent(room);
+    } catch (e) {
+        // This can happen though if we get an unencoded string and it contains
+        // some characters that look like an encoded entity, but it's not.
+        // But in this case we're fine goin on...
+    }
+
+    // Only decoded strings can be lowercased properly.
+    // We use locale aware lower case to properly convert national characters.
+    room = room.toLocaleLowerCase();
+
+    // But we still need to (re)encode it.
+    room = encodeURIComponent(room);
```

Comment:
I was recommended to explicitly call .normalize on the string with the "NKFC" option. Normalizing would take care of characters that look the same but aren't the same value being converted to the actual same value.

Reply:
But maybe the lowercasing already handles that? Then I got nothing. Looks good.

I had initially asked when this was reported if this should be fixed on the client or the server. Did you ever get a response to that? Were you told this to be fixed on the client?

Reply:
You’re right. Normalizing with NFC would help indeed. While a given representation may look the same in Unicode, it might be expressed differently, so the extra normalize step is A Good Thing. Good call.

Reply:
@zbettenbuk, I had initially asked when this was reported if this should be fixed on the client or the server. Did you ever get a response to that? Were you told this to be fixed on the client?